toolkit/scripts/check_required_kernel_configs.py

```diff
@@ -0,0 +1,134 @@
+        config_json_data = json.load(req_config_json_file)
+        if kernel not in config_json_data:
+            print(f"Kernel {kernel} not found in {json_file}")
+            print(f"Please provide required configs for {kernel} in {json_file}")
+            print(f"Exiting...")
+            return None
+        required_configs_data = config_json_data[kernel]['required-configs']
+
+    #incorrect configs is map: {config: (newValue, expectedValue, comment, PR)}
+    incorrect_configs = {}
+
+    # go through required configs
+    for config_option, value in required_configs_data.items():
+        # check for arch
+        if arch not in value['arch']:
+            continue
+        # check if required config is present with correct value
+        found = False
+        # check for required config in each input config line (without extra _VALUE)
+        for line in input_config_data.split('\n'):
+            if f"{config_option}=" in line or f"{config_option} is not set" in line:
+                for val in value['value']:
+                    if val in line and val != "":
+                        found = True
+                        break
+                # config was found but value is not correct
+                # mark as found and add to incorrect_configs
+                if not found:
+                    incorrect_configs[config_option] = (IncorrectConfig(config_option, line.split(config_option)[1].replace('=',''), value['value'], value['comment'], value['PR']))
+                    found = True
+        if not found:
+            # check if config can be missing
+            if "" not in value['value']:
+                incorrect_configs[config_option] = (IncorrectConfig(config_option, line, value['value'], value['comment'], value['PR']))
+
+    return incorrect_configs
+
+def print_verbose(json_file, kernel, arch, results):
+    with open(json_file, 'r') as req_config_json_file:
+        data = json.load(req_config_json_file)
+        required_configs_data = data[kernel]['required-configs']
+    print_data = [["Option", "Required Arch", "Expected Value", "Comment"]]
+    for config_option, value in required_configs_data.items():
+        if arch in value['arch']:
+            if config_option in results:
+                print_data.append([config_option, value['arch'], value['value'], f"FAIL: Unexpected value: {results[config_option].newValue} See: {value['PR']}"])
+            else:
+                print_data.append([config_option, value['arch'], value['value'], "OK"])
+    # Calculate maximum width for each column
+    column_widths = [max(len(str(row[i])) for row in print_data) for i in range(len(print_data[0]))]
+
+    # Print columns
+    for j, row in enumerate(print_data):
+        for i, column in enumerate(row):
+            print(str(column).ljust(column_widths[i] + 2), end='')
+        print()
+        if j == 0:
+            print("-------------------------------------------------------------------------------")
```

Comment:
I don't know what's going on here, it's already past my bedtime, I trust we want this and the output is just stunning to behold.:)